src/resolvePath.js

```diff
@@ -80,13 +94,10 @@ function resolvePathFromAliasConfig(sourcePath, currentFile, opts) {
 const resolvers = [
   resolvePathFromAliasConfig,
   resolvePathFromRootConfig,
+  resolvePathFromRelativeConfig,
 ];
 
 export default function resolvePath(sourcePath, currentFile, opts) {
-  if (isRelativePath(sourcePath)) {
```

Comment:
Do we really need to remove this condition?

I mean.. Resolving a relative path is only about potentially adding the custom extension, right?

Reply:
If we want to make sure the relative path is checked first we can change the order of the resolvers to have the relativeResolver first. Which is logically equivalent to having an if statement that explicitly runs the relative resolver.

I'm not opposed to that, but I think this is more a question of where we want the if checks of whether to run the resolver. For example, rather than have the array of resolvers, we could just have multiple if statements that decide whether to run each resolver. But I do think we should be consistent. My leaning is towards having the if checks in the resolvers.

Reply:
Agreed. Sorry, it's been a while since I worked in this area of the code and I didn't read it properly. `resolvePath` is called for every "resolvers" so indeed we have to remove it.

But then, should we actually run `resolvePathFromAliasConfig` and `resolvePathFromRootConfig` on relative paths? I don't think so, right? (cc @fatfisz)
Should relative paths only run `resolvePathFromRelativeConfig` to potentially resolve the custom extension?

Reply:
Possibly somebody might want a relative path alias "e.g. ./lib -> some/other/path/lib". So there might be a small case for running aliases for relative paths. resolving from root, probably not.

Depending on the order we put the resolvers, the others won't run. e.g. with the way I have it now, if the path is relative, it will always run resolvePathFromRelativeConfig first, and so will never reach the other two resolvers.

Reply:
i agree with @amosyuen . it might slow down resolution but perhaps we should add an exclude/include option so we can exclude/include specific directories/regexes, ....

for the android, ios case you probably have different babel configs, one that looks for ios.js & .js and another one that does for android.js and .js